packages/components/src/local/organisms/setting-channels/types/props.d.ts

```diff
@@ -24,4 +24,32 @@ export type MessageGroupProps = {
   value: string[]
 }
 
```

Comment:
This is a minor suggestion @lebaphi - but how about declaring each props in the type for that channel?  So, `CollabChannelProps` would be in `collab.tsx`?

Reply:
Hmm, maybe that's not possible.  I'll look at the code if/when I get chance.

Reply:
I'm not clear the purpose of this @IanMayo ?

Reply:
Sure - happy to explain :-)

This is the idea..  We have 4 types of channel,  one is mapping.  If we make changes to the mapping channel configs, I hope that only `mapping` specific files would get edited.   Then we _know_ that we can't accidentally break another type of channel. Currently, the `mapping` props are in the same file as the other channel types.  So, adding a new mapping prop, and "tidying up the file" could mean that we accidentally change the props for another channel type.

Reply:
ah I see what you mean. Yep, the mapping type is in the same file with other channel types but it's a separate type object, we can change it without effects to others. But we can move it into the channel file. I will update

Reply:
You'll see that because I am crazy, I have tried to use generics to declare the props.  To do this I had to switch from `type` to `interfaces`.  This doesn't seem to introduce any compiler warnings, and things seem to work ok.

It would still be _neat_ to declare the types in channel-specific files, if we can.